File: typescript/packages/aholo-sdk-lux3d/src/lux3d-client.ts
import { createGatewayClient, type AholoClientConfig, type AholoGatewayClient } from '@manycore/aholo-sdk-core';

import { ImgTo3dResource } from './resources/img-to-3d.js';
import { MaterialTransferResource } from './resources/material-transfer.js';
import { TasksResource } from './resources/tasks.js';
import { TextTo3dResource } from './resources/text-to-3d.js';

/**
 * Aholo Lux3D API client.
 *
 * Stainless-style resource access:
 * ```ts
 * const lux3d = createLux3dClient({ apiKey: '...' })
 * await lux3d.imgTo3d.create({ img: '...' })
 * await lux3d.imgTo3d.createFromFile('/path/to/image.png')
 * await lux3d.textTo3d.create({ prompt: '...' })
 * await lux3d.materialTransfer.create({ img: '...', meshUrl: '...' })
 * await lux3d.tasks.retrieve(taskId)
 * await lux3d.tasks.waitFor(taskId)
 * ```
 */
export class Lux3dClient {
  readonly imgTo3d: ImgTo3dResource;
  readonly textTo3d: TextTo3dResource;
  readonly materialTransfer: MaterialTransferResource;
  readonly tasks: TasksResource;

  private readonly gateway: AholoGatewayClient;

  constructor(config: AholoClientConfig = {}) {
    this.gateway = createGatewayClient(config);
    this.imgTo3d = new ImgTo3dResource(this.gateway, config.region);
    this.textTo3d = new TextTo3dResource(this.gateway, config.region);
    this.materialTransfer = new MaterialTransferResource(this.gateway, config.region);
    this.tasks = new TasksResource(this.gateway, config.region);
  }

}

export function createLux3dClient(config: AholoClientConfig = {}): Lux3dClient {
  return new Lux3dClient(config);
}

File: python/packages/aholo-sdk-lux3d/src/manycore/aholo_sdk_lux3d/lux3d_client.py
from __future__ import annotations

from typing import Optional

from manycore.aholo_sdk_core import AholoClientConfig, create_gateway_client

from .resources.img_to_3d import ImgTo3dResource, _file_to_data_url
from .resources.material_transfer import MaterialTransferResource
from .resources.part_split import PartSplitResource
from .resources.tasks import TasksResource
from .resources.text_to_3d import TextTo3dResource

# Re-export for convenience: from manycore.aholo_sdk_lux3d import file_to_data_url
file_to_data_url = _file_to_data_url


class Lux3dClient:
    """
    Aholo Lux3D API client.

    Stainless-style resource access::

        lux3d = Lux3dClient(api_key="...")
        task_id = lux3d.img_to_3d.create(img="data:image/png;base64,...")
        task_id = lux3d.img_to_3d.create_from_file("/path/to/image.png")
        task_id = lux3d.text_to_3d.create(prompt="A wooden chair")
        task_id = lux3d.material_transfer.create(img="...", mesh_url="https://...")
        task_id = lux3d.part_split.create(glb_url="https://example.com/model.glb")
        result  = lux3d.tasks.retrieve(task_id)
        result  = lux3d.tasks.wait_for(task_id)
    """

    def __init__(self, config: Optional[AholoClientConfig] = None) -> None:
        cfg = config or AholoClientConfig()
        gateway = create_gateway_client(cfg)
        region: str = cfg.region or "cn"
        self.img_to_3d = ImgTo3dResource(gateway, region)
        self.text_to_3d = TextTo3dResource(gateway, region)
        self.material_transfer = MaterialTransferResource(gateway, region)
        self.part_split = PartSplitResource(gateway, region)
        self.tasks = TasksResource(gateway, region)


def create_lux3d_client(config: Optional[AholoClientConfig] = None) -> Lux3dClient:
    return Lux3dClient(config)
